source/ext/patches/fluentd/in_syslog.patch

```diff
@@ -9,12 +37,79 @@
  
      def configure(conf)
        super
-@@ -251,7 +253,7 @@
+@@ -226,7 +232,7 @@
+         end
+ 
+         ## from receive_data_default
+-        pri = record.delete('pri'.freeze) unless pri
++        pri = record.delete('pri') unless pri
+ 
+         facility = FACILITY_MAP[pri >> 3]
+         priority = PRIORITY_MAP[pri & 0b111]
+@@ -244,14 +250,23 @@
+     private
+ 
+     def listen(callback)
+-      log.info "listening syslog socket on #{@bind}:#{@port} with #{@protocol_type}"
++
+       if @protocol_type == :udp
++        log.info "listening syslog socket on #{@bind}:#{@port} with #{@protocol_type}"
+         @usock = SocketUtil.create_udp_socket(@bind)
+         @usock.bind(@bind, @port)
          SocketUtil::UdpHandler.new(@usock, log, @message_length_limit, callback, @resolve_hostname)
-       else
+-      else
++      elsif @protocol_type == :tcp
++        log.info "listening syslog socket on #{@bind}:#{@port} with #{@protocol_type}"
          # syslog family add "\n" to each message and this seems only way to split messages in tcp stream
 -        Coolio::TCPServer.new(@bind, @port, SocketUtil::TcpHandler, log, "\n", callback, @resolve_hostname)
 +        Coolio::TCPServer.new(@bind, @port, SocketUtil::TcpHandler, log, @delimiter, callback, @resolve_hostname)
```

Comment:
if we already have a patch to add delimiter, is there any conflict by changing it here as well

Reply:
I have updated the same patch, this patch will include both delimiter + unix support